src/index.ts

```diff
@@ -18,6 +18,15 @@ export interface Collection<K, V> extends Map<K, V> {
 	constructor: CollectionConstructor;
 }
 
+/**
+ * A utility type for merging type of collections with two seperate value types.
+ *
+ * @internal
+ */
+export type ValueMerged<K, V, C2 extends Collection<K, unknown>> = C2 extends Collection<K, infer O>
```

Comment:
Shouldn't this type not be exported if it's internal?

Reply:
I don't think so considering other internal types are still exported.